File: workspace/src/banhang/SalesManagement.Domain/Entities/Product.cs
namespace SalesManagement.Domain.Entities;

public class Product
{
    public Guid Id { get; private set; }
    public string Name { get; private set; }
    public string Sku { get; private set; }
    public decimal Price { get; private set; }
    public string Description { get; private set; }
    public string Unit { get; private set; } = "Cái"; // Mặc định
    
    public Guid? CategoryId { get; private set; }
    public Category? Category { get; private set; }

    private Product() { } // EF Core constructor

    public Product(string name, string sku, decimal price, string description, string unit = "Cái", Guid? categoryId = null)
    {
        Id = Guid.NewGuid();
        Name = name;
        Sku = sku;
        Price = price;
        Description = description;
        Unit = unit;
        CategoryId = categoryId;
    }

    public void Update(string name, string sku, decimal price, string description, string unit = "Cái", Guid? categoryId = null)
    {
        Name = name;
        Sku = sku;
        Price = price;
        Stock = stock;
        Description = description;
        Unit = unit;
        CategoryId = categoryId;
    }
}

File: workspace/src/bandoan/TapHoa/taphoa-storefront/src/domain/entities/Product.ts
export interface Category {
  id: string;
  name: string;
  slug: string;
  iconName: string;
  itemCount: number;
}

export interface Product {
  id: string;
  name: string;
  slug: string;
  price: number;
  originalPrice?: number;
  discountPercent?: number;
  categoryId: string;
  categoryName: string;
  image: string;
  images?: string[]; // Gallery images (additional product photos)
  unit: string; // e.g. '500g', '1kg', 'Chai 1L', 'Hộp', 'Túi 500g'
  stock: number;
  rating: number;
  soldCount: number;
  description: string;
  origin?: string; // Nguồn gốc xuất xứ: Việt Nam, Nhật Bản, v.v.
  isFlashSale?: boolean;
  isPopular?: boolean;
}
